testbed/eks/lib/addons/fluxv2.ts

```diff
@@ -0,0 +1,110 @@
+import cdk = require('@aws-cdk/core');
+import eks = require('@aws-cdk/aws-eks');
+import * as yaml from 'js-yaml';
+import * as request from 'sync-request';
+
+export interface FluxV2Props extends cdk.StackProps {
+    cluster: eks.Cluster;
+    fluxRepos: Repository[];
+    secretName: string;
+    fluxVersion: string;
+}
+export class Repository {
+    repoUrl: string;
+    repoBranch: string;
+    repoPath: string;
+
+    constructor(repoUrl: string, repoBranch?: string, repoPath?: string) {
+        if (!repoBranch) {
+            this.repoBranch = "main"
+        } else {
+            this.repoBranch = repoBranch
+        }
+        if (!repoPath) {
+            this.repoPath = "test/workflows"
+        } else {
+            this.repoPath = repoPath
+        }
+        this.repoUrl = repoUrl
+    }
+}
+class FluxRelease {
+    private fluxVersion: string;
+    private manifestUrl: string;
+    public installManifest: any;
+
+    constructor(version: string) {
+        this.fluxVersion = version;
+        this.manifestUrl = `https://github.com/fluxcd/flux2/releases/download/${this.fluxVersion}/install.yaml`
+    }
+
+    public getUrl(): string {
+        return this.manifestUrl;
+    }
+
+    public getManifest(): any {
+        this.installManifest = yaml.loadAll(
+            request.default('GET', this.manifestUrl)
```

Comment:
The eks construct docs recommend `const manifest = yaml.safeLoadAll(request('GET', manifestUrl).getBody());` which I think is pretty clean.